pkg/cache/remote.go

```diff
@@ -22,11 +22,22 @@ type RemoteCache struct {
 // RemoteURL to hold remote host
 type RemoteURL string
 
+// Insecure for RPC remote host
+type Insecure bool
+
 // NewRemoteCache is the factory method for RemoteCache
-func NewRemoteCache(url RemoteURL, customHeaders http.Header) cache.ArtifactCache {
+func NewRemoteCache(url RemoteURL, insecure Insecure, customHeaders http.Header) (cache.ArtifactCache, error) {
 	ctx := client.WithCustomHeaders(context.Background(), customHeaders)
-	c := rpcCache.NewCacheProtobufClient(string(url), &http.Client{})
-	return &RemoteCache{ctx: ctx, client: c}
+
+	httpTransport := http.DefaultTransport.(*http.Transport).Clone()
+	httpTransport.TLSClientConfig.InsecureSkipVerify = bool(insecure)
+
+	httpClient := &http.Client{
+		Transport: httpTransport,
+	}
```

Comment:
What if we define the HTTP client directly?

```
	client := &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				InsecureSkipVerify: true,
			},
		},
	}

Reply:
This way, we would lose the other defaults of `http.Transport`.

When I clone the default, I only change the `InsecureSkipVerify` setting.

Reply:
Changed that in the latest commits.